app.go

```diff
@@ -0,0 +1,55 @@
+package httpcache
+
+import (
+	"errors"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/darkweak/souin/cache/surrogate/providers"
+	"github.com/darkweak/souin/cache/types"
+	"github.com/darkweak/souin/configurationtypes"
+)
+
+// SouinApp contains the whole Souin necessary items
+type SouinApp struct {
+	*DefaultCache
+	Provider         types.AbstractProviderInterface
+	SurrogateStorage providers.SurrogateInterface
+	API              configurationtypes.API `json:"api,omitempty"`
+	LogLevel         string                 `json:"log_level,omitempty"`
+}
+
+func init() {
+	caddy.RegisterModule(SouinApp{})
+}
+
+// Provision implements caddy.Provisioner
+func (s *SouinApp) Provision(_ caddy.Context) error {
```

Comment:
You should at least inject the zap logger in Souin isn't it?

Also, I don't understand how Souin is started. It's a side effect?

Reply:
You may want different loglevel on each directive.